### PR TITLE
[SPARK-45045][SS] Revert back the behavior of idle progress for StreamingQuery API from SPARK-43183

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This PR proposes to revert back the behavior of idle progress for StreamingQuery API from [SPARK-43183](https://issues.apache.org/jira/browse/SPARK-43183), to avoid breakage of tests from 3rd party data sources.

### Why are the changes needed?

We indicated that the behavioral change from SPARK-43183 broke many tests in 3rd party data sources.
(Short summary of SPARK-43183: we changed the behavior of idle progress to only provide idle event callback, instead of making progress update callback as well as adding progress for StreamingQuery API to provide as recent progresses/last progress.)

The main rationale of SPARK-43183 was to avoid making progress update callback for idle event, which had been confused users. That is more about streaming query listener, and not necessarily had to change the behavior of StreamingQuery API as well.

### Does this PR introduce _any_ user-facing change?

Yes, but the user-facing change is technically reduced before this PR, as we revert back the behavioral change partially from SPARK-43183, which wasn't released yet. 

### How was this patch tested?

Modified tests. Also manually ran 3rd party data source tests which were broken with Spark 3.5.0 RC which succeeded with this change.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
Will have a separate PR for branch-3.5 to reduce a hop in case there is merge conflict.

**Comment by HeartSaVioR** [comment]
DISCLAIMER: I wanted to revert relevant commits and re-work on supporting idle event which would be easier for reviewers to look at, but this was also touched with Spark connect effort hence very hard to revert.

**Comment by HeartSaVioR** [comment]
cc. @xuanyuanking

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala [hunk 1]
```diff
@@ -172,96 +188,102 @@ trait ProgressReporter extends Logging {
       currentTriggerLatestOffsets != null)
     currentTriggerEndTimestamp = triggerClock.getTimeMillis()
 
-    if (hasExecuted) {
-      val executionStats = extractExecutionStats(hasNewData)
-      val processingTimeMills = currentTriggerEndTimestamp - currentTriggerStartTimestamp
-      val processingTimeSec = Math.max(1L, processingTimeMills).toDouble / MILLIS_PER_SECOND
+    val executionStats = extractExecutionStats(hasNewData, hasExecuted)
```
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala:191]: Context: L191-L251 is technically revert of SPARK-43183.

**Comment by anishshri-db** [comment]
@HeartSaVioR - revert itself looks fine. Is there a way to check if lastProgress outputs the same result before SPARK-43183 and with this change ?

**Review by anishshri-db — APPROVED** [review]

**Review by jerrypeng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
Thanks for reviewing! Merging to master.

**Pull request closed without merging**